src/__experimental__/components/checkbox/checkbox-group.component.js

```diff
@@ -7,26 +7,53 @@ import FormField from '../form-field';
 
 const CheckboxGroup = (props) => {
   const {
+    id,
     children,
-    groupName,
     hasError,
     hasWarning,
-    hasInfo
+    hasInfo,
+    onChange,
+    onBlur,
+    value,
+    defaultValue
   } = props;
 
-  const groupLabelId = `${groupName}-label`;
+  const defaultChecked = (defaultValue !== undefined) ? defaultValue : [];
+  const checkboxGroupValue = (value !== undefined) ? value : defaultChecked;
+  const groupLabelId = `${id}-label`;
+  const [checkedValue, setCheckedValue] = useState(checkboxGroupValue);
 
-  const buttons = React.Children.map(children, (child) => {
-    const handleChange = (ev) => {
-      child.props.onChange(ev);
+  const onChangeProp = useCallback((e) => {
+    const checkedCheckboxes = [...checkedValue];
+    const checkedIndex = checkedCheckboxes.indexOf(e.target.value);
+
+    if (checkedIndex === -1) {
+      checkedCheckboxes.push(e.target.value);
+    } else {
+      checkedCheckboxes.splice(checkedIndex, 1);
+    }
+
+    setCheckedValue(checkedCheckboxes);
+
+    e.target = {
+      id,
```

Comment:
I need to think about this more, but we've come to the consensus that we need to provide `name` in all events. This is complicated by the example above when there are two children with different names.

I think it is best to remove the `onChange` functionality from the `CheckboxGroup` component. We may re-introduce it for controlled components when we introduce `formik`. But for now, because the event object is inconsistent and I can't think of a way of determining the correct name. Let's remove the `onChange` helper from `CheckboxGroup`. This is not to say that this component does not have a `onChange` applied to the cloned child - we just don't want to call the user provided `onChange`.

I think having `onBlur` on the `CheckboxGroup` will be confusing, does it blur when all of them lose focus, what if you select one checkbox then another. To prevent it from being ambiguous, let's remove this.